django-nonrel/ocl/oclapi/models.py

```diff
@@ -37,12 +41,29 @@ class BaseModel(models.Model):
     created_by = models.TextField()
     updated_by = models.TextField()
     is_active = models.BooleanField(default=True)
+    is_being_saved = False
     extras = DictField(null=True, blank=True)
+    extras_are_encoded = False
+    extras_are_decoded = False
```

Comment:
Do we really need both extras_are_encoded  and extras_are_decoded ?

Reply:
Since the values are decoded in the setter (and the setter is called very many times from very many places including during saves, fetches, views, etc.) I believe the safest way to ensure values are not multi-encoded or multi-decoded is to use separate variables for encoded and decoded rather than trying to track every instance an on/off switch might be toggled.

Reply:
I think either way you need to code it correctly since you should not end up in an inconsistent state with extras_are_encoded = True and extras_are_decoded = True. Not sure how using 2 variables makes it safer. Are you saying it is okay to be in such a state with both values set to True or False?